beater/config/sampling.go

```diff
@@ -17,17 +17,84 @@
 
 package config
 
+import (
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/apm-server/elasticsearch"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
 // SamplingConfig holds configuration related to sampling.
 type SamplingConfig struct {
 	// KeepUnsampled controls whether unsampled
 	// transactions should be recorded.
 	KeepUnsampled bool `config:"keep_unsampled"`
+
+	// Tail holds tail-sampling configuration.
+	Tail *TailSamplingConfig `config:"tail"`
+}
+
+// TailSamplingConfig holds configuration related to tail-sampling.
+type TailSamplingConfig struct {
+	Enabled bool `config:"enabled"`
+
+	DefaultSampleRate     float64               `config:"default_sample_rate" validate:"min=0, max=1"`
+	ESConfig              *elasticsearch.Config `config:"elasticsearch"`
+	Interval              time.Duration         `config:"interval" validate:"min=1"`
+	IngestRateDecayFactor float64               `config:"ingest_rate_decay" validate:"min=0, max=1"`
+	StorageDir            string                `config:"storage_dir"`
+	StorageGCInterval     time.Duration         `config:"storage_gc_interval" validate:"min=1"`
+	TTL                   time.Duration         `config:"ttl" validate:"min=1"`
```

Comment:
What exactly does `min=1` mean for `time.Duration`?

Reply:
`min=1` is synonymous with `min=1s`. What I actually want to check is that it's positive, but go-ucfg's "positive" validator actually means "non-negative". Anyway, perhaps it would be clearer if I changed it to `min=1s`?

Reply:
yes I think `min=1s` would be easier to read

Reply:
done